### PR TITLE
Ethereum address sanity check when generating from public key derived from private or mnemonic

### PR DESCRIPTION
**Pull request by vforvalerio87**
Verifying whether the address generated from the public key is already prefixed with 0x, in which case it's not added twice

### CODE REVIEW

File: snet_cli/identity.py [hunk 1]
```diff
@@ -47,8 +47,8 @@ def __init__(self, w3, private_key):
                                                   curve=ecdsa.SECP256k1,
                                                   hashfunc=hashlib.sha256).get_verifying_key()
 
-        self.address = self.w3.toChecksumAddress(
-            "0x" + self.w3.sha3(hexstr=public_key.to_string().hex())[12:].hex())
+        self.address = self.w3.sha3(hexstr=public_key.to_string().hex())[12:].hex()
+        self.address = self.w3.toChecksumAddress(self.address) if self.address.startswith("0x") else self.w3.toChecksumAddress("0x" + self.address)
```
Comment (astroseger) [inline snet_cli/identity.py]: You probably should directly use web3.eth.accounts.privateKeyToAccount.

Comment (vforvalerio87) [reply snet_cli/identity.py]: That's JavaScript: https://web3js.readthedocs.io/en/1.0/web3-eth-accounts.html#privatekeytoaccount

I couldn't find the same in Python.

**Comment by vforvalerio87** [comment]
> Am I right that you add this check because of some bug on your machine. Is it possible to add some tests (functional or unit) which cover this bug?

Actually this already fails for me with our current tests, see:
https://circleci.com/gh/singnet/snet-cli/278#tests/containers/0

**Comment by vforvalerio87** [comment]
So apparently now this is a requirement for #255 

Tests already cover this in case it fails

Should we merge this in?

File: snet_cli/identity.py (hunk 1 above)
Comment (astroseger) [reply snet_cli/identity.py]: @vforvalerio87 it took 30sec to find web3py alternative!
```python
import web3
print(web3.eth.Account.privateKeyToAccount(self.private_key).address)
```
please use it

Comment (astroseger) [reply snet_cli/identity.py]: (it is actually not so easy to find this function in web3py. I found it in the source code of web3py. But I think we should use it... )

**Pull request merged**